packages/data-ingestion/lib/index.ts

```diff
@@ -0,0 +1,104 @@
+
+const logger = getLogger('BigQueryClient');
+
+interface RunScriptRow {
+    executionType: string;
+    internalConnectionId: number | undefined;
+    connectionId: string;
+    accountId: number | undefined;
+    scriptName: string;
+    scriptType: string;
+    environmentId: number;
+    providerConfigKey: string;
+    status: string;
+    syncId: string;
+    content: string;
+    runTimeInSeconds: number;
+    createdAt: number;
+}
+
+class BigQueryClient {
+    private client: BigQuery;
+    private datasetName: string;
+    private tableName: string;
+
+    constructor({ datasetName, tableName }: { datasetName: string; tableName: string }) {
+        this.client = new BigQuery();
+        this.tableName = tableName;
+        this.datasetName = datasetName;
+    }
+
+    static async createInstance({ datasetName, tableName }: { datasetName?: string; tableName: string }) {
+        const instance = new BigQueryClient({
+            datasetName: datasetName || 'raw',
+            tableName
+        });
+        await instance.initialize();
+        return instance;
+    }
+
+    private async initialize() {
+        try {
+            if (isCloud) {
+                await this.createDataSet();
+                await this.createTable();
+            }
+        } catch (e) {
+            logger.error('Error initializing', e);
+        }
+    }
+
+    private async createDataSet() {
+        const dataset = this.client.dataset(this.datasetName);
+        const [exists] = await dataset.exists();
+        if (!exists) {
+            await this.client.createDataset(this.datasetName);
+        }
+    }
+
+    private async createTable() {
+        const table = this.client.dataset(this.datasetName).table(this.tableName);
+        const [exists] = await table.exists();
+        if (!exists) {
+            await table.create({
+                schema: {
+                    fields: [
+                        { name: 'executionType', type: 'STRING' },
+                        { name: 'internalConnectionId', type: 'INTEGER' },
+                        { name: 'connectionId', type: 'STRING' },
+                        { name: 'accountId', type: 'INTEGER' },
+                        { name: 'scriptName', type: 'STRING' },
+                        { name: 'scriptType', type: 'STRING' },
+                        { name: 'environmentId', type: 'INTEGER' },
+                        { name: 'providerConfigKey', type: 'STRING' },
+                        { name: 'status', type: 'STRING' },
+                        { name: 'syncId', type: 'STRING' },
+                        { name: 'content', type: 'STRING' },
+                        { name: 'runTimeInSeconds', type: 'FLOAT' },
+                        { name: 'createdAt', type: 'INTEGER' }
+                    ]
+                }
+            });
+        }
+    }
+
+    public async insert(data: RunScriptRow, tableName?: string) {
```

Comment:
Since we are not enforcing type and schema matching, should insert simply accept any record<string, string|number>?

Reply:
Won't work as then there would be a type mismatch between the two.